plenum/test/common/test_digest_validation.py

```diff
@@ -94,17 +122,38 @@ def wait_one_batch(before):
     assert Request(**json.loads(req1)).digest in idrs
 
 
-def test_suspicious_primary_send_same_request_with_different_signatures(
-        looper, txnPoolNodeSet, sdk_pool_handle, two_requests):
-    def malicious_dynamic_validation(self, request: Request):
-        self.execute_hook(NodeHooks.PRE_DYNAMIC_VALIDATION, request=request)
-
-        operation = request.operation
-        req_handler = self.get_req_handler(txn_type=operation[TXN_TYPE])
-        req_handler.validate(request)
+def test_parts_of_nodes_have_same_request_with_different_signatures(
+        looper, txnPoolNodeSet, sdk_pool_handle, two_requests, sdk_wallet_stewards, tconf):
+    req1s, req2s = two_requests
+    req1 = Request(**json.loads(req1s))
+    req2 = Request(**json.loads(req2s))
+    for node in txnPoolNodeSet[0:2]:
+        req_state = node.requests.add(req1)
+        req_state.propagates['Alpha'] = req1
+        req_state.propagates['Beta'] = req1
+        req_state.propagates['Gamma'] = req1
+        req_state.propagates['Delta'] = req1
+        node.tryForwarding(req1)
+
+    for node in txnPoolNodeSet[2:4]:
+        req_state = node.requests.add(req2)
+        req_state.propagates['Alpha'] = req2
+        req_state.propagates['Beta'] = req2
+        req_state.propagates['Gamma'] = req2
+        req_state.propagates['Delta'] = req2
+        node.tryForwarding(req2)
+
+    req1s = sdk_send_signed_requests(sdk_pool_handle, [req1s])
+    sdk_get_and_check_replies(looper, req1s)
```

Comment:
Please check that all nodes have the same data.
Also we expect here that nodes having `req2` only will request Propagates for req1, so it makes sense to check it as well.

Reply:
ok